mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -125,6 +125,60 @@ void mlir::sycl::SYCLDialect::printType(
   }
 }
 
+llvm::Optional<llvm::StringRef>
+mlir::sycl::SYCLDialect::findMethod(mlir::TypeID BaseType,
+                                    llvm::StringRef MethodName) const {
+  return syclCtx.findMethod(BaseType, MethodName);
```

Comment:
This could call the `MethodRegistry::findMethod` rather than calling `SYCLContext::findMethod` (the latter is just a forwarder).